### PR TITLE
Transition to Spezi Data Pipeline package

### PR DESCRIPTION
**Pull request by Vicbi**
# *Transition to Spezi Data Pipeline package*

## :gear: Release Notes 
- Imported [spezi_data_pipeline](https://test.pypi.org/project/spezi-data-pipeline/0.1.0b1/) package as a dependency and removed duplicated code.
- Renamed `ECGDataPipelineTemplate` folder to `ecg_data_manager`
- Ensured the “No more ECG data to review” message is cleared from the screen after reviewing all unreviewed recordings and new initials are selected.
- Updated the Diagnosis dropdown list to replace “Normal” with “Normal Sinus Rhythm”.
- Implemented a filter for review status using the reviewer’s initials, so previously reviewed recordings by the current reviewer will not be displayed. 
- Enriched the output .CSV file by adding columns for full name, date of birth, participant group (kid or adult), and diagnosis details (reviewer’s initials, selected diagnosis, tracing quality, and notes).
- Developed a secondary notebook for managing reviewed ECG recordings, enabling filtering by participant group, user ID, date, and Apple Watch classifications  (e.g., AFib). 
- Included an indication of the subject’s group (child/adult) and their age next to the recording under consideration.
- Added pylint and black formatter  
- Refactored scripts and added documentation

## :books: Documentation
Updated documentation to include instructions on the second notebook (`ECGExplorer.ipynb`).

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: CONTRIBUTORS.md
Comment (PSchmiedmayer) [inline CONTRIBUTORS.md:1]: 🎉

File: .github/workflows/build-and-test.yml
Comment (PSchmiedmayer) [inline .github/workflows/build-and-test.yml:1]: Thanks for updating and improving the merge checks @Vicbi!

File: ecg_data_manager/ECGExplorer.ipynb
Comment (PSchmiedmayer) [inline ecg_data_manager/ECGExplorer.ipynb:1]: Nice to see how compact these have become 🚀

File: scripts/upload_codes.py
Comment (PSchmiedmayer) [inline scripts/upload_codes.py:1]: Thanks for adding all the documentation here!!

File: ecg_data_manager/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+firebase_admin
+pandas
+numpy
+matplotlib
+ipywidgets
+fhir.resources
+--extra-index-url https://test.pypi.org/simple/
+spezi-data-pipeline==0.1.0b1
```
Comment (PSchmiedmayer) [inline ecg_data_manager/requirements.txt:8]: 🚀

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for all the improvements; great to see the all the improvements in here and get this merged in the main branch 🚀

File: CONTRIBUTORS.md
Comment (Vicbi) [reply CONTRIBUTORS.md:1]: Thank you @PSchmiedmayer for reviewing the latest changes! Excited to have this merged into the main branch!

**Pull request merged**